doc/introduction/mid_circuit_measurements.rst

```diff
@@ -0,0 +1,530 @@
+
+.. code-block:: python
+
+    dev = qml.device("default.qubit")
+
+    @qml.qnode(dev)
+    def my_qnode(x, y):
+        qml.RY(x, wires=0)
+        qml.CNOT(wires=[0, 1])
+        m_0 = qml.measure(1, reset=False, postselect=None)
+
+        qml.cond(m_0, qml.RY)(y, wires=0)
+        return qml.probs(wires=[0]), qml.expval(m_0)
+
+See the following sections for details on 
+:func:`~.pennylane.measure`, :func:`~.pennylane.cond`, and statistics
+of mid-circuit measurements, as well as information about simulation
+strategies and how to configure them.
+Additional information can be found in the documentation of the individual
+methods. Also consider our
+`Introduction to mid-circuit measurements
+<https://pennylane.ai/qml/demos/tutorial_mcm_introduction/>`__,
+`How to collect statistics of mid-circuit measurements
+<https://pennylane.ai/qml/demos/tutorial_how_to_collect_mcm_stats/>`__,
+and `How to create dynamci circuits with mid-circuit measurements
+<https://pennylane.ai/qml/demos/tutorial_how_to_create_dynamic_mcm_circuits/>`__.
+
+Resetting wires
+***************
+
+Wires can be reused after making mid-circuit measurements. Moreover, a measured wire can be
+reset to the :math:`|0 \rangle` state by setting ``reset=True`` in :func:`~.pennylane.measure`:
+
+.. code-block:: python3
+
+    dev = qml.device("default.qubit", wires=3)
+
+    @qml.qnode(dev)
+    def func():
+        qml.PauliX(1)
+        m_0 = qml.measure(1, reset=True)
+        qml.PauliX(1)
+        return qml.probs(wires=[1])
+
+Executing this QNode:
+
+.. code-block:: pycon
+
+    >>> func()
+    tensor([0., 1.], requires_grad=True)
+
+Postselecting mid-circuit measurements
+**************************************
+
+PennyLane also supports postselecting on mid-circuit measurement outcomes by specifying the
+``postselect`` keyword argument of :func:`~.pennylane.measure`. By default, postselection
+discards outcomes that do not match the ``postselect`` argument.
+For example, specifying ``postselect=1`` is equivalent to projecting the state vector onto
+the :math:`|1\rangle` state, i.e., disregarding all outcomes where :math:`|0\rangle` is measured:
+
+.. code-block:: python3
+
+    dev = qml.device("default.qubit")
+
+    @qml.qnode(dev)
+    def func(x):
+        qml.RX(x, wires=0)
+        m_0 = qml.measure(0, postselect=1)
+        return qml.sample(wires=0)
+
+By postselecting on ``1``, we only consider results that measured the outcome ``1``.
+Executing this QNode with 10 shots yields
+
+.. code-block:: pycon
+
+    >>> func(np.pi / 2, shots=10)
+    array([1, 1, 1, 1, 1, 1, 1])
+
+Note that only 7 samples are returned. This is because samples that do not meet the postselection criteria are
+discarded. This behaviour can be customized, see the section
+:ref:`"Configuring mid-circuit measurements" <mcm_config>`.
+
+.. note::
+
+    Currently, postselection is only supported on :class:`~.pennylane.devices.DefaultQubit`.
```

Comment:
Thanks, good catch! Moved it to the simulation techniques part, in particular to the deferred measurements section, because it is a feature support detail.